From a8b78f2790e6eac08c5fe01905f02ed441b5ca29 Mon Sep 17 00:00:00 2001
From: Meron Cohen <mecohen@habana.ai>
Date: Thu, 08 May 2025 18:56:40 +0300
Subject: [PATCH] [SW-227605] g2e: update init bis cache

Support G2E BIS revision when initializing BIS cache.
In case we can't read the BIS revision from boot info and eFuse,
then assume it is G2E BIS revision (rev 05). Otherwise, update
it according to what's read (adding G2E BIS revision use-case).

Change-Id: I3682ec60fb312c776ebc41e263dc7f2249c32860
Signed-off-by: Meron Cohen <mecohen@habana.ai>
(cherry picked from commit af9ca7de53b481a488b9ed8f8cb55ec8570ed4c1)
---

diff --git a/drivers/misc/common_utils/gaudi/gaudi2/common_utils.c b/drivers/misc/common_utils/gaudi/gaudi2/common_utils.c
index 2226ae3..92efd6b 100644
--- a/drivers/misc/common_utils/gaudi/gaudi2/common_utils.c
+++ b/drivers/misc/common_utils/gaudi/gaudi2/common_utils.c
@@ -128,8 +128,8 @@
 		LOG_WRN("Failed to read security fuse info, try to read from fuse");
 		err = efuse_read_word(efuse_dev, FUSE_CONF_BITS_WORD_IDX, &val);
 		if (err) {
-			LOG_ERR("Failed to read bis compliant fuse, assuming true, revD");
-			bis_compliant_cache.bis_rev = GAUDI2_BIS_REV_D;
+			LOG_ERR("Failed to read bis compliant fuse, assuming true, revE");
+			bis_compliant_cache.bis_rev = GAUDI2_BIS_REV_E;
 			goto bis_cache_set_valid;
 		}
 	} else {
@@ -137,11 +137,16 @@
 	}
 
 	is_bis_compliant = !!(val & EFUSE_BIS_COMPLIANT_MASK);
-	if (is_bis_compliant)
-		bis_compliant_cache.bis_rev = (val & EFUSE_ENABLE_BIS_D_COMPLIANT) ?
-						GAUDI2_BIS_REV_D : GAUDI2_BIS_REV_C;
-	else
+	if (is_bis_compliant) {
+		if (val & EFUSE_ENABLE_BIS_E_COMPLIANT)
+			bis_compliant_cache.bis_rev = GAUDI2_BIS_REV_E;
+		else if (val & EFUSE_ENABLE_BIS_D_COMPLIANT)
+			bis_compliant_cache.bis_rev = GAUDI2_BIS_REV_D;
+		else
+			bis_compliant_cache.bis_rev = GAUDI2_BIS_REV_C;
+	} else {
 		bis_compliant_cache.bis_rev = GAUDI2_BIS_NONE;
+	}
 
 bis_cache_set_valid:
 	bis_compliant_cache.is_valid = true;
